### ISSUE TITLE
PLR1714: Repeated equality comparison to the same literal value shouldn't convert to a set

### ISSUE THREAD

**Issue by injust** [astral-sh/ruff#21692]
### Summary

Observed this in and split it out of https://github.com/astral-sh/ruff/issues/21690.

```python
def test(foo: str) -> None:
    if foo == "bar" or foo == "bar":
        pass
```

https://play.ruff.rs/40871aec-3cf3-48fd-b42f-76f1b8c4118f

PLR1714 changes this to:

```python
def test(foo: str) -> None:
    if foo in {"bar", "bar"}:
        pass
```

If the values being compared are literals, the equality comparisons should be merged instead of converting to a set.

**Comment by MichaReiser** [comment]
I'm leaning towards not flagging `PLR1714` if all members are equal because fixing it to a single equal isn't what the rule is testing for. Detecting that the second equal is unnecessary is something that falls into a generic unreachable rule
